kolibri/plugins/device/assets/src/views/AvailableChannelsPage/index.vue

```diff
@@ -171,7 +171,18 @@
       ]),
       allChannels() {
         // Need to de-duplicate channels in case user enters same token twice, etc.
-        return uniqBy([...this.newUnlistedChannels, ...this.availableChannels], 'id');
+        const allPossibleChannels = uniqBy(
```

Comment:
I'm not sure this is quite correct yet. We actually want `this.newUnlistedChannels` to appear at the beginning of the list. Here's the scenario that you might not have seen yet (it's a brand new feature in Kolibri)

1. Go to the Import View.
1. Click the "Select Entire Channels Instead" link at the top (will put a checkbox next to all the channels).
1. Click "Import with token" and use the same token I gave you in a comment.
1. When you click "submit", 1) "slides, slides, slides" channel should be at the top, 2) the other installed channels should be at the bottom (note if "slides" was already installed, don't show again, then 3) the other uninstalled channels.

Reply:
So I think this is very close, just make sure that `...this.newUnlistedChannels` is at the beginning of the returned array, `installedChannelsWithResourecs` is in the middle, and the filtered array is at the very end :+1:

Reply:
Ah, thanks for shedding some light on that! I've moved `this.newUnlistedChannels` to the top of the list (image appended to original comment). 

> if "slides" was already installed, don't show again

Does this mean that you don't expect "slides" to be shown in the 'import' view if it was already installed? If so, from what I can tell it's included in `this.installedChannelsWithResources` and I'm not sure if that should be the case then.

Reply:
If you have these channels installed

1. Public Channel 1
2. Public Channel 2
3. Unlisted Channel 

Then the import view should show

1. Public Channel 1
2. Public Channel 2
3. Unlisted Channel 
4. (the other not-installed channels)

But if you enter the token for unlisted channel, it should show

3. Unlisted Channel (now at the top, and not repeated below)
1. Public Channel 1
2. Public Channel 2
4. (the other not-installed channels)

Reply:
Thanks for the clarification! I believe those cases are all handled now; I've added another screenshot to the original post.